test/osmedile/intellij/stringmanip/swap/SwapCharacterToFromIntActionTest.java

```diff
@@ -0,0 +1,154 @@
+
+
+    private static String escapeJava(String str) {
+        return escapeJavaStyleString(str);
+    }
+
+    private static String escapeJavaStyleString(String str) {
+        if (str == null) {
+            return null;
+        } else {
+            try {
+                StringWriter writer = new StringWriter(str.length() * 2);
+                escapeJavaStyleString(writer, str);
+                return writer.toString();
+            } catch (IOException var3) {
+                var3.printStackTrace();
+                return null;
+            }
+        }
+    }
+
+    private static void escapeJavaStyleString(Writer out, String str) throws IOException {
+        if (out == null) {
+            throw new IllegalArgumentException("The Writer must not be null");
+        } else if (str != null) {
+            int sz = str.length();
+
+            for (int i = 0; i < sz; ++i) {
+                char ch = str.charAt(i);
+                if (ch > 127) {
+                    out.write("\\" + oct(ch));
+                } else if (ch < 32) {
+                    switch (ch) {
+                        case '\b':
+                            out.write(92);
+                            out.write(98);
+                            break;
+                        case '\t':
+                            out.write(92);
+                            out.write(116);
+                            break;
+                        case '\n':
+                            out.write(92);
+                            out.write(110);
+                            break;
+                        case '\u000b':
+                        default:
+                            out.write("\\" + oct(ch));
+                            break;
+                        case '\f':
+                            out.write(92);
+                            out.write(102);
+                            break;
+                        case '\r':
+                            out.write(92);
+                            out.write(114);
+                    }
+                } else {
+                    switch (ch) {
+                        case '"':
+                            out.write(92);
+                            out.write(34);
+                            break;
+                        case '\'':
+                            out.write(92);
+                            out.write(39);
+                            break;
+                        case '/':
+                            out.write(92);
+                            out.write(47);
+                            break;
+                        case '\\':
+                            out.write(92);
+                            out.write(92);
+                            break;
+                        default:
+                            out.write(ch);
+                    }
+                }
+            }
+        }
+    }
+
+    private static String oct(char ch) {
+        return Integer.toString(ch, 8);
+    }
```

Comment:
that's quite a lot of duplicate code, why not just make SwapCharacterToFromIntAction#escapeJava protected and call it?

Reply:
didn't want to export a function that had little to do with the primary purpose of the class and since it was a one-off i figured  "meh". i only wish the producers of the 'StringEscapeUtils' utilities would allow the escape encoding to be configurable.

I'll make `escapeJava` package-private and update the PR.